backend/molgenis-emx2-datamodels/src/main/java/org/molgenis/emx2/datamodels/profiles/SchemaFromProfile.java

```diff
@@ -17,72 +22,155 @@
 
 public class SchemaFromProfile {
 
-  private String yamlFileLocation;
-  private Profiles profiles;
-  private Character separator;
+  private final Profiles profiles;
 
   public SchemaFromProfile(String yamlFileLocation) {
     if (!yamlFileLocation.endsWith(".yaml")) {
       throw new MolgenisException("Input YAML file name must end in '.yaml'");
     }
-    this.yamlFileLocation = yamlFileLocation;
     InputStreamReader yaml =
         new InputStreamReader(
             Objects.requireNonNull(
-                AbstractDataLoader.class
-                    .getClassLoader()
-                    .getResourceAsStream(this.yamlFileLocation)));
+                AbstractDataLoader.class.getClassLoader().getResourceAsStream(yamlFileLocation)));
     try {
       this.profiles = new ObjectMapper(new YAMLFactory()).readValue(yaml, Profiles.class);
     } catch (Exception e) {
       throw new MolgenisException(e.getMessage(), e);
     }
-    this.separator = ',';
   }
 
   public Profiles getProfiles() {
     return profiles;
   }
 
   public SchemaMetadata create() throws MolgenisException {
+    String sharedModelsDir = File.separator + "_models" + File.separator + "shared";
+    String specificModelsDir = File.separator + "_models" + File.separator + "specific";
     List<Row> keepRows = new ArrayList<>();
+    try {
+      keepRows.addAll(getProfilesFromAllModels(sharedModelsDir));
```

Comment:
didn't test, but can it be specific models and shared models have same name? We can say this is problem of the profile editor so no showstopper.